hudi-common/src/main/java/org/apache/hudi/common/table/timeline/MetadataConversionUtils.java

```diff
@@ -244,18 +246,44 @@ public static Option<HoodieCommitMetadata> getHoodieCommitMetadata(HoodieTableMe
     return Option.of(TimelineUtils.getCommitMetadata(hoodieInstant, timeline));
   }
 
-  public static org.apache.hudi.avro.model.HoodieCommitMetadata convertCommitMetadata(
-          HoodieCommitMetadata hoodieCommitMetadata) {
-    ObjectMapper mapper = new ObjectMapper();
-    // Need this to ignore other public get() methods
-    mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
-    org.apache.hudi.avro.model.HoodieCommitMetadata avroMetaData =
-            mapper.convertValue(hoodieCommitMetadata, org.apache.hudi.avro.model.HoodieCommitMetadata.class);
+  /**
+   * Convert commit metadata from json to avro.
+   */
+  public static <T extends SpecificRecordBase> T convertCommitMetadata(HoodieCommitMetadata hoodieCommitMetadata) {
+    if (hoodieCommitMetadata instanceof HoodieReplaceCommitMetadata) {
+      return (T) convertReplaceCommitMetadata((HoodieReplaceCommitMetadata) hoodieCommitMetadata);
+    }
+    hoodieCommitMetadata.getPartitionToWriteStats().remove(null);
```

Comment:
Can we add some doc why there is a `null` key in the map.

Reply:
i'm not sure why this was there even previously. I just kept the same logic.
https://github.com/apache/hudi/blob/1862f17be5b99f51b61c22c777b3d39361f4abd5/hudi-common/src/main/java/org/apache/hudi/common/model/HoodieCommitMetadata.java#L226-L229